### PR TITLE
Amit/fix total allocationfor quota

### PR DESCRIPTION
**Pull request by amitbose327**

### CODE REVIEW

File: aaad/mesos.py [hunk 1]
```diff
@@ -51,6 +51,26 @@ def get_tasks(master_url):
...
+    tasks_count = mesos.get_tasks_counts(master_url.rstrip('/'))
+    limit = int(sum(tasks_count.values()))
+    url = '{}/tasks?limit={}'.format(master_url.rstrip('/'), limit)
+    cache_key = 'get_tasks' + master_url
+    tasks = cache.get(cache_key)
+    if tasks:
+        return tasks
+    tasks = {}
+    resp = requests.get('{}'.format(url))
+    if not resp.status_code // 100 == 2:
```
Comment (jhulten) [inline aaad/mesos.py]: Would raising an exception provide more context?
http://docs.python-requests.org/en/master/user/quickstart/#errors-and-exceptions

File: aaad/mesos.py [hunk 2]
```diff
@@ -51,6 +51,26 @@ def get_tasks(master_url):
...
+        logger.error('Got response {} from {}'.format(resp.status_code, url))
+        raise ValueError('Got a non-2xx response ({}) from {}.'.format(resp.status_code, url))
+    data = resp.json()
+    for task in data['tasks']:
+        if task['state'] in ['TASK_RUNNING', 'TASK_STAGING', 'TASK_STARTING']:
+            tasks[task['id']] = { 'cpus': task['resources']['cpus'], 'mem': task['resources']['mem'], 'disk': task['resources']['disk'] }
+    cache[cache_key] = tasks
```
Comment (jhulten) [inline aaad/mesos.py]: If processes die and are recreated, how is the cache updated?

Comment (amitbose327) [inline aaad/mesos.py]: In case a process dies, the cache will start from new.

File: aaad/mesos.py [hunk 3]
```diff
@@ -51,6 +51,26 @@ def get_tasks(master_url):
     cache[cache_key] = tasks
     return tasks
 
+def get_task_ids_and_resources(master_url):
+    tasks_count = mesos.get_tasks_counts(master_url.rstrip('/'))
+    limit = int(sum(tasks_count.values()))
+    url = '{}/tasks?limit={}'.format(master_url.rstrip('/'), limit)
+    cache_key = 'get_tasks' + master_url
```
Comment (ankanm) [inline aaad/mesos.py]: Change to `... 'get_task_ids_and_resources' +...`

**Comment by ankanm** [comment]
LGTM

**Pull request merged**